lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -47,12 +47,34 @@ sub _git_clone_all ($job, $clones) {
 
     # iterate clones sorted by path length to ensure that a needle dir is always cloned after the corresponding casedir
     for my $path (sort { length($a) <=> length($b) || $a cmp $b } keys %$clones) {
-        my $url = $clones->{$path};
+        my $url = $clones->{$path} // '';
         die "Don't even think about putting '..' into '$path'." if $path =~ /\.\./;
+
+        my $server;
+        if ($url =~ /gitlab/) { $server = 'gitlab'; }
+        elsif ($url =~ /github/) { $server = 'github'; }
+        else { $server = 'other'; }
```

Comment:
```suggestion
        if ($url =~ /gitlab/) { $server = 'gitlab' }
        elsif ($url =~ /github/) { $server = 'github' }
        else { $server = 'other' }
```

Reply:
went with the oneliner you suggested later on